zebROS_ws/src/behaviors/src/path_intake_actionlib_server.cpp

```diff
@@ -0,0 +1,165 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+
+#include <behavior_actions/GamePiecePickup.h>
+#include <nav_msgs/Path.h>
+#include <behavior_actions/IntakeAction.h>
+#include <path_follower_msgs/PathAction.h>
+
+// This action lib server will first generate a path with game_piece_path_gen, then run that path using PathAction, then intake a game piece using IntakeAction.
+
+void waitForActionlibServer(T &action_client, double timeout, const std::string &activity);
+
+class PathIntakeAction{
+  private:
+    ros::NodeHandle nh_;
+    actionlib::SimpleActionServer<behaviors_actions::IntakeAction> as_;
+    std::string action_name_;
+    behaviors_actions::PathIntakeFeedback feedback_;
+    behaviors_actions::PathIntakeResult result_;
+
+  public:
+    PathIntakeAction(std::string name):
```

Comment:
Nitpicking, but the name can be passed as a const std::string &name. Passing it as a reference (using &) will prevent the code from making a copy of it. And marking it const makes sure that the code doesn't accidentally change the variable.

It doesn't make much difference here (the constructor is just called once, so the extra copy doesn't add much overhead) but it is a good habit to get in to for other cases where it might matter. For example, passing images or other large data structures around in code which is called much more frequently.